pyproject.toml

```diff
@@ -26,13 +26,12 @@ requires-python = ">=3.7"
 dev = [
     "black",
     "mypy",
-    "flake8-isort",
-    "Flake8-pyproject",
     "pipdeptree",
     "pre-commit",
     "pydata-sphinx-theme>=0.12",
     "pytest",
     "pytest-cov",
+    "ruff",
```

Comment:
There is a ruff pre-commit hook which bundles ruff, and the ruff vscode extension bundles ruff. I guess we could do without this as a dependency if we use the pre-commit hook? Although then we wouldn't be able to run `ruff --fix` on the commandline...

Thoughts?

Reply:
My main thought behind this was indeed around `ruff --fix` - I decided to disable auto-fix in the editor (because it can auto-fix quite a lot of things, some of which might be a little surprising). So then leaving the option for people to run `ruff --fix` on the command line is nice. And I think generally I like the idea that we can run our tools "standalone" (i.e. not via pre-commit) - makes it easier to quickly check for linting errors if editing via `ssh` without a GUI for example.

I'm definitely open to arguments on this one though, can be persuaded either way if there are strong opinions.

Reply:
I think include it. I expect I will set format on save in my user settings, so I won't use the CLI much, but I am sure there will be a time and it will be annoying if it isn't there.

Reply:
Ok, let's leave it in